opal/datatype/opal_datatype_memcpy.h

```diff
@@ -11,10 +11,35 @@
  * $HEADER$
  */
 
+#include "opal/runtime/opal.h"
+
 #ifndef OPAL_DATATYPE_MEMCPY_H_HAS_BEEN_INCLUDED
 #define OPAL_DATATYPE_MEMCPY_H_HAS_BEEN_INCLUDED
 
-#define MEMCPY( DST, SRC, BLENGTH ) \
-    memcpy( (DST), (SRC), (BLENGTH) )
+#define TYPED_MEMCPY( DST, SRC, BLENGTH, ALIGN, TYPE) \
+    if( ((BLENGTH) == sizeof(TYPE)) && (0 == ((ALIGN) & (sizeof(TYPE) - 1))) ) { \
+        for( size_t _i = 0; _i < ((BLENGTH) / sizeof(TYPE)); _i++ ) \
```

Comment:
why looping here?
since sizeof(TYPE) = BLENGTH, we are sure _i is always and only equal to 0.
Or may be I missed something?

Reply:
My final goal was to have some kind of Duff's device going from the largest to the smallest predefined type. I will update the code.